### PR TITLE
Feat/fct 18131/create insight timeoff

### PR DESCRIPTION
**Pull request by RamProg**
## 🚪 Why?

Part of the new Profile work, we are creating insights

## 🔑 What?

This PR creates a Timeoff widget

## 🏡 Context

https://www.figma.com/design/9pLsRzdinid0ha0qRWta2D/Employee-Profile?node-id=604-7408&t=dFJO9AADh1bYqZcy-0

## 📸 Visual proof

image

### CODE REVIEW

File: src/playground/Insights/ui/event.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { cn } from "@/lib/utils"
+import { forwardRef } from "react"
+
+interface EventProps {
+  title: string
+  length: number
+  lengthUnit: string
+  from: string
+  until: string
```
Comment (sauldom102) [inline src/playground/Insights/ui/event.tsx:9]: Don't you think we should leave this more open to `title` & `subtitle` props only? And then in Factorial we just form these texts based on the data that we have

**Review by sauldom102 — COMMENTED** [review]
These are shifts in the mobile app, they have some icons too. Don't we have those in web?

Screenshot 2024-08-27 at 09 42 42

File: src/playground/Insights/Tasks/index.tsx [hunk 2]
```diff
@@ -67,49 +67,48 @@ export const TasksInsight = forwardRef<HTMLDivElement, TasksInsightProps>(
...
+        <WidgetContainer
+          header={{
+            title,
+            link: { title: linkTitle, url: linkUrl },
+          }}
+        >
+          <div className="grid grid-cols-3">
+            {taskCategories.map(({ label, count, icon, color }) => (
+              <Indicator
+                label={label}
```
Comment (sauldom102) [inline src/playground/Insights/Tasks/index.tsx:79]: We're missing a `key` here

File: src/playground/Insights/Tasks/index.tsx [hunk 3]
```diff
@@ -67,49 +67,48 @@ export const TasksInsight = forwardRef<HTMLDivElement, TasksInsightProps>(
...
+                count={count}
+                icon={icon}
+                color={color}
+              />
+            ))}
+          </div>
+          <Separator />
+          <div className="flex flex-col gap-3">
+            {!tasks?.length ? (
+              <p>{emptyStateText ?? "No tasks"}</p>
```
Comment (sauldom102) [inline src/playground/Insights/Tasks/index.tsx:89]: I think this text should be mandatory as a prop and not assign any default here

File: src/playground/Insights/Timeoff/index.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                    until={item.until}
+                    color={item.color}
+                    isPending={true}
+                  />
+                ))
+            )}
+          </div>
+          <Separator />
+          <p className="mb-4 font-medium text-muted-foreground">
+            Current and upcoming
```
Comment (sauldom102) [inline src/playground/Insights/Timeoff/index.tsx:75]: This text should come localized in a prop right? 🤔

File: src/playground/Insights/Tasks/index.tsx (hunk 3 above)
Comment (RamProg) [reply src/playground/Insights/Tasks/index.tsx:89]: Makes sense

File: src/playground/Insights/Tasks/index.tsx (hunk 2 above)
Comment (RamProg) [reply src/playground/Insights/Tasks/index.tsx:79]: Good catch

**Comment by RamProg** [comment]
> These are shifts in the mobile app, they have some icons too. Don't we have those in web?
> 
> Screenshot 2024-08-27 at 09 42 42

I'm not sure, we would need to ask @lily-factorial 
Looks like we can always ask them later, right?

**Comment by sauldom102** [comment]
> > These are shifts in the mobile app, they have some icons too. Don't we have those in web?
> > Screenshot 2024-08-27 at 09 42 42
> 
> I'm not sure, we would need to ask @lily-factorial Looks like we can always add them later, right?

Sure! It just took my attention cause I recently did this component in mobile

**Pull request merged**